ecs-cli/modules/cli/local/network/network.go

```diff
@@ -0,0 +1,217 @@
+// Copyright 2015-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Package network provides functionality to setup and teardown the ECS local network.
+package network
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/client"
+
+	"github.com/docker/docker/api/types/container"
+
+	"github.com/docker/docker/api/types/filters"
+
+	"github.com/sirupsen/logrus"
+	"golang.org/x/net/context"
+
+	"github.com/docker/docker/api/types/network"
+
+	"github.com/docker/docker/api/types"
+)
+
+// IdempotentLocalNetworkStarter groups Docker functions to create the ECS local network and
+// the Local Container Endpoints container if they don't exist.
+type IdempotentLocalNetworkStarter interface {
+	idempotentNetworkCreator
+	idempotentContainerStarter
+}
+
+type idempotentNetworkCreator interface {
+	NetworkInspect(ctx context.Context, networkID string, options types.NetworkInspectOptions) (types.NetworkResource, error)
+	NetworkCreate(ctx context.Context, name string, options types.NetworkCreate) (types.NetworkCreateResponse, error)
+}
+
+type idempotentContainerStarter interface {
+	ContainerList(ctx context.Context, options types.ContainerListOptions) ([]types.Container, error)
+	ContainerCreate(ctx context.Context, config *container.Config, hostConfig *container.HostConfig, networkingConfig *network.NetworkingConfig, containerName string) (container.ContainerCreateCreatedBody, error)
+	ContainerStart(ctx context.Context, containerID string, options types.ContainerStartOptions) error
+}
+
+// Configuration for the ECS local network.
+const (
+	// EcsLocalNetworkName is the name of the network created for local ECS tasks to join.
+	EcsLocalNetworkName = "ecs-local-network"
+
+	// Range of IP addresses that containers in the network get assigned.
+	ecsLocalNetworkSubnet = "169.254.170.0/24"
+)
+
+// Configuration for the Local Endpoints container.
+const (
```

Comment:
are/will we consume `amazon/amazon-ecs-local-container-endpoints` package as a dependency? I feel like these constants would exist in that package and we should reference them from there if possible.